tournesol/views/comparison.py

```diff
@@ -47,15 +46,38 @@ def list_comparison(self, request, *args, **kwargs):
             return Response('No comparison found', status=404)
         return Response(ComparisonSerializer(comparisons, many=True).data)
 
+    def get_video(self, request_data):
+        """
+            Tries to fetch the video from Tournesol's database and returns an appropriate error
+            message if it is not found
+        """
+        try:
+            video_id = request_data['video_id']
+        except KeyError:
+            return None, Response(
+                {"detail": "Incorrect format for video ids"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        video_query = Video.objects.filter(video_id=video_id)
+        if not video_query:
+            return None, Response(
+                {"detail": f"The video with id {video_id} does not exist"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        return video_query.first(), None 
+
```

Comment:
Maybye you could use the Video serializer which is doing the same thing
I don't to which extend the video may exist or not in the db.
```suggestion
    def get_video(self, request_data):
        """
            Tries to fetch the video from Tournesol's database and returns an appropriate error
            message if it is not found
        """
        video = Video.objects.get(video_id = request_data.video_id)
        serializer = Video(snippet,)
        return Response(serializer.data)

```

Reply:
The sinppet that you propose when trying to access the `video_id` attribute would raise a server error if the input data does not have this attribute. The user of the API would then receive a Server 500 error that will not be very helpful. This is why I wrote the `try...except` above.

Videos may not exist in the database